src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -120,7 +125,9 @@ public async Task<string> GetDocumentAsync(string address, CancellationToken can
                 if (response.IsSuccessStatusCode)
                     return responseContent;
 
-                 unsuccessfulHttpResponseException = new IOException(LogHelper.FormatInvariant(LogMessages.IDX20807, address, response, responseContent));
+                unsuccessfulHttpResponseException = new IOException(LogHelper.FormatInvariant(LogMessages.IDX20807, address, response, responseContent));
```

Comment:
Should we create a different exception name for this? If we did, we could have StatusCode and ResponseContent as properties on it.

Not sure if we'd consider changing the exception type a breaking change though.

Reply:
I  create a new exception(GetOidcConfigurationException) in SAL and it has StatusCode and ResponseContent as properties there

Reply:
Would it make more sense for Wilson to throw that exception directly? Is there anything in it that is first party only?

Reply:
George has mentioned in teams channel, it might not the best solution to parse the response from AAD in Wilson since Wilson is IDP-agnostic.